src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,88 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
 
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+
+        if (clazz == null) {
+            throw new RuntimeException("There is no implementation for class "
+                    + interfaceClazz.getName());
+        }
+
+        componentAnnotationChecker(clazz);
+
+        Object classImplementationInstance = createNewInstance(clazz);
+
+        List<Field> fields = List.of(clazz.getDeclaredFields());
+
```

Comment:
remove redundant empty lines
```suggestion
        componentAnnotationChecker(clazz);
        Object classImplementationInstance = createNewInstance(clazz);
        List<Field> fields = List.of(clazz.getDeclaredFields());
```